sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md

```diff
@@ -0,0 +1,116 @@
+# Implement VSLAM algorithm based on KubeEdge
+
+This project aims to develop a Visiual SLAM application on an edge-side robot and implementing compute offloading to reduce the compute and memory resource pressure on edge-side.
+
+## Prerequisites
+
+1. Kubeedge  1.10.1+
+2. Edgemesh  1.11.0+
+
+## Steps to reproduce
+
+1. Make sure your cluster is set up and [edgemesh](https://kubeedge.io/en/docs/advanced/edgemesh/) is enabled.
+
+2. Clone the community/sig-robotics respository.
+
+   ```bash
+   git clone https://github.com/kubeedge/community
+   ```
+
+3. Prepare your RGBD dataset or RGBD-sensors on the edge-side.
+
+   > We use [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) and its format in our development and test process.
+
+4. Check your work node's name and modify the deployment.yaml file.
+
+   > For example, we have a master node and two work nodes, named edge01 and edge02
+
+   `cloud-deployment.yaml`
+
+   ```yaml
+   spec:
+     template:
+       spec:
+         nodeName: edge02
+   ```
+
+   `edge-deployment.yaml`
+
+   ```yaml
+   spec:
+     template:
+       spec:
+         nodeName: edge01
+   ```
+
+   Check the location of the downloaded  [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) and modify `edge-deployment.yaml`
+
+   ```yaml
+   spec:
+     template:
+       spec:
+         - name: data-mount
+           hostPath: 
+             path: [your_absolute_path]/workspace/data
+   ```
+
+5. Make sure you have replace the `PATH_TO_SEQUENCE_FOLDER` ,`ASSOCIATIONS_FILE` in [edge-deployment.yaml](edge-deployment.yaml) and `TUN_NAME` in [cloud-deployment.yaml](cloud-deployment.yaml)to your own parameters.
+
+   ```yaml
+   env:
+   - name: PATH_TO_SEQUENCE_FOLDER
+     value: /workspace/data/YOUR_DATA_FOLDER
+   - name: ASSOCIATIONS_FILE
+     value: /workspace/Edgeslam/Examples/RGB-D/YOUR_CONFIG.yaml
+   ```
+
+   ```yaml
+   env:
+   - name: TUM_NAME
+     value: TUM2
+   ```
+
+6. Create the applications deployment.
+
+   ```bash
+   kubectl apply -f edge-deployment.yaml
+   kubectl apply -f cloud-deployment.yaml
+   ```
+
+7. If your edge-side node support X11 GUI, you will see a frame windows and a map window. The trajectory will be saved under `/workspace/edgeslam/Examples`  folder.
+
+## Validation
+
+We tested origin [ORB_SLAM2]() and ours on the  [TUM datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download).
```

Comment:
The works we has use for reference including [ORB_SLAM2](https://github.com/raulmur/ORB_SLAM2), and combine the performance on [TUM datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download). We illustrate the overall framework of ours in the Figure: